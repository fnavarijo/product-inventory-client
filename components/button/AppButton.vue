<template>
  <component
    :is="as"
    :to="to"
    class="py-4 px-10 w-max font-bold transition duration-100 ease-linear"
    :class="color"
  >
    <slot />
  </component>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'AppButton',
  props: {
    as: {
      type: String,
      default: 'button',
      validator: value => ['nuxt-link', 'button'].includes(value),
    },
    to: {
      type: String,
      default: '/',
    },
    color: {
      type: String,
      default: 'primary',
      validator: value => ['primary', 'secondary'].includes(value),
    },
  },
});
</script>

<style scoped>
.primary {
  @apply bg-primary-500 text-white;
}

.secondary {
  @apply bg-white text-dark-gray-600 border-4 border-gray-800;
}

.primary:hover,
.secondary:hover {
  @apply text-white bg-dark-gray-600;
}
</style>
