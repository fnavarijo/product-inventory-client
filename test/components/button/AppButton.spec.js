import { mount, RouterLinkStub } from '@vue/test-utils';

import AppButton from '@/components/button/AppButton.vue';

describe('AppButton component', () => {
  it('should emit event on click', async () => {
    const wrapper = mount(AppButton);

    wrapper.vm.$emit('click');

    await wrapper.vm.$nextTick();

    expect(wrapper.emitted().click).toBeTruthy();
  });

  it('should render component as button as default', () => {
    const wrapper = mount(AppButton);

    const buttonElement = wrapper.find('button');

    expect(buttonElement.exists()).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render component as NuxtLink when passed prop', () => {
    const wrapper = mount(AppButton, {
      stubs: {
        'nuxt-link': RouterLinkStub,
      },
      propsData: {
        as: 'nuxt-link',
      },
    });

    const buttonElement = wrapper.find('button');

    expect(buttonElement.exists()).toBeFalsy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render button component with secondary colors', () => {
    const wrapper = mount(AppButton, {
      propsData: {
        color: 'secondary',
      },
    });

    const buttonElement = wrapper.find('button');

    expect(buttonElement.exists()).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
