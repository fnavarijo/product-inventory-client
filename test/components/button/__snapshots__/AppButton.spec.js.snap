// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton component should render button component with secondary colors 1`] = `<button to="/" class="py-4 px-10 w-max font-bold transition duration-100 ease-linear secondary"></button>`;

exports[`AppButton component should render component as NuxtLink when passed prop 1`] = `<a class="py-4 px-10 w-max font-bold transition duration-100 ease-linear primary"></a>`;

exports[`AppButton component should render component as button as default 1`] = `<button to="/" class="py-4 px-10 w-max font-bold transition duration-100 ease-linear primary"></button>`;
